onnxruntime/python/tools/quantization/save_activations.py

```diff
@@ -0,0 +1,153 @@
+        op_types_for_saving = []
+    saver = CalibraterBase(onnx_model, op_types_to_calibrate=op_types_for_saving)
+    model = clone_model_with_shape_infer(saver.model)  # type: ModelProto
+    tensors, _ = saver.select_tensors_to_calibrate(model)
+    reshape_shape_name = "LinearReshape_" + str(time.time())
+    reshape_shape = numpy_helper.from_array(numpy.array([-1], dtype=numpy.int64), reshape_shape_name)
+    model.graph.initializer.append(reshape_shape)
+
+    for tensor_name in tensors:
+        reshape_output = tensor_name + _tensor_save_postfix_
+        reshape_node = onnx.helper.make_node(
+            "Reshape",
+            inputs=[tensor_name, reshape_shape_name],
+            outputs=[reshape_output],
+            name=reshape_output,
+        )
+        model.graph.node.append(reshape_node)
+        vinfo = helper.make_tensor_value_info(reshape_output, TensorProto.FLOAT, [1])
+        model.graph.output.append(vinfo)
+    return model
+
+
+def run_collect_activations(
+    augmented_model: str,
+    input_reader: CalibrationDataReader,
+    session_options=None,
+    execution_providers: Optional[List[str]] = None,
+) -> dict:
+    r"""Run augmented model and collect activations tensors
+
+    Parameters
+    ----------
+    augmented_model : str
+        Path to augmented model created by augment_model_save_tensors()
+    input_reader : CalibrationDataReader
+        Logic for reading input for the model, augmented model have the same
+        input with the original model.
+    session_options :
+        Optional OnnxRuntime session options for controlling model run.
+        By default graph optimization is turned off
+    execution_providers : List[str]
+        Optional execution providers for running the model. CPU EP is used
+        by default.
+
+    Returns
+    -------
+    A dictionary where the key is tensor name and values are list of tensors from each batch
+    """
+
+    if session_options is None:
+        session_options = onnxruntime.SessionOptions()
+        session_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_DISABLE_ALL
+    if execution_providers is None:
+        execution_providers = ["CPUExecutionProvider"]
+
+    infer_session = onnxruntime.InferenceSession(
+        augmented_model,
+        sess_options=session_options,
+        providers=execution_providers,
+    )
+
+    intermediate_outputs = []
+    for input_d in input_reader:
+        intermediate_outputs.append(infer_session.run(None, input_d))
+        if len(intermediate_outputs) == 0:
+            raise ValueError("No data is collected while running augmented model!")
+
+    output_dict = {}
+    output_info = infer_session.get_outputs()
+    for batch in intermediate_outputs:
+        for output, output_data in zip(output_info, batch):
+            if output.name.endswith(_tensor_save_postfix_):
+                oname = output.name[0 : len(output.name) - _tensor_save_postfix_len_]
```

Comment:
```suggestion
                oname = output.name[: -_tensor_save_postfix_len_]
```

nit: no need to specify the start position when it is 0; we can also use negative indexes to save a call to len(output.name)